dragonfly/engines/base/recobs.py

```diff
@@ -77,15 +78,20 @@ def notify_begin(self):
                                     " method of recognition observer %s: %s"
                                     % (observer, e))
 
-    def notify_recognition(self, words):
+    def notify_recognition(self, words, rule, node):
         for observer in self._observers:
             try:
                 if hasattr(observer, "on_recognition"):
-                    observer.on_recognition(words)
+                    (arg_names, _, kwargs_name, _) = inspect.getargspec(observer.on_recognition)
```

Comment:
Lines like this are causing a lot of deprecation warnings when running the tests in Python 3. I think we should use `inspect.getargspec()` for Python 2.7 and `inspect.getfullargspec()` in Python 3.

Reply:
Fixed. There's another `getargspec()` deprecation warning in `action_function.py` that I'll fix separately.